src/index.ts

```diff
@@ -0,0 +1,288 @@
+	const repo = payload.repository.name
+	const htmlUrl = payload.issue.html_url
+
+	if (ALLOWED_ORGS.includes(owner) === false) {
+		info(`Ignoring ${htmlUrl} from ${owner}`)
+		return
+	}
+
+	const commentId = payload?.comment?.id as number
+	const body = payload?.comment?.body || ''
+
+	const author = payload.issue?.user?.login || ''
+	const prNumber = payload.issue?.number
+	const prTitle = payload.issue?.title || `Backport of PR #${prNumber}`
+
+	const authOctokit = new Octokit({ auth: await getAuthToken(installationId) })
+
+	// Ignoring comments on issues that are not PRs
+	if (!payload?.issue?.pull_request) {
+		return
+	}
+
+	// Check if the comment is a backport request
+	if (body.trim().startsWith(COMMAND_PREFIX)) {
+		// Check if the author is at least a collaborator
+		const commentAuthor = payload?.comment?.user.login
+		const authorAssociation = payload?.comment?.author_association
+		if (!authorAssociation || authorAssociation === 'NONE') {
+			info(`Ignoring comment from non-collaborator: ${commentAuthor}}`)
+			return
+		}
+
+		let branch: string
+		let commits: string[] = []
+
+		const isForcedRequest = body.trim().startsWith(COMMAND_PREFIX + '!')
+		const isFullRequest = body.trim().startsWith(isForcedRequest ? COMMAND_PREFIX + '!' + TO_SEPARATOR : COMMAND_PREFIX + TO_SEPARATOR)
+		const isClosed = payload.issue?.state === 'closed'
+		const isMerged = typeof payload.issue?.pull_request?.merged_at === 'string'
+
+		if (isClosed && !isMerged) {
+			addReaction(authOctokit, { owner, repo, commentId } as Task, Reaction.THUMBS_DOWN)
+			error(`Ignoring comment on closed but unmerged PR ${htmlUrl}`)
+			return
+		}
+
+		// Extract the commits and branch from the payload
+		try {
+			commits = extractCommitsFromPayload(body)
+			branch = extractBranchFromPayload(body)
+		} catch (e) {
+			// Add a confused reaction to the comment to indicate that we failed to understand it
+			addReaction(authOctokit, { owner, repo, commentId } as Task, Reaction.CONFUSED)
+			error(`Failed to extract commits and branch from payload: \`${body}\` on ${htmlUrl}`)
+			return
+		}
+
+		// Start processing the request
+		try {
+			// If we have no commits, and the request did specify some commits
+			// then something went wrong.
+			// /backport `5e83e97 to stable28` means we backport 5e83e97 to stable28
+			// /backport to stable28 means we backport all commits from this PR to stable28
+			if (commits.length === 0 && !isFullRequest) {
+				throw new Error('No commits found in payload')
+			}
+
+			if (isFullRequest) {
+				info(`\nReceived full backport request to \`${branch}\``)
+				info(`├ Fetching commits from PR ${htmlUrl}...`)
+				commits = await getCommitsForPR(authOctokit, owner, repo, prNumber)
+			} else {
+				info(`\nReceived partial backport request to \`${branch}\``)
+			}
+
+			// PR info
+			if (isForcedRequest && isMerged) {
+				info(`├ PR is merged, but force flag is present, starting backport right away`)
```

Comment:
do you mean to log this?
```suggestion
			if (isForcedRequest && !isMerged) {
				info(`├ PR is not merged, but force flag is present, starting backport right away`)
```